planner/core/resolve_indices.go

```diff
@@ -213,12 +226,25 @@ func (p *PhysicalMergeJoin) ResolveIndices() (err error) {
 	shallowColSlice := make([]*expression.Column, p.schema.Len())
 	copy(shallowColSlice, p.schema.Columns)
 	p.schema = expression.NewSchema(shallowColSlice...)
-	for i := 0; i < colsNeedResolving; i++ {
-		newCol, err := p.schema.Columns[i].ResolveIndices(mergedSchema)
-		if err != nil {
-			return err
+	foundCnt := 0
+	// The two column sets are all ordered. And the colsNeedResolving is the subset of the mergedSchema.
```

Comment:
so this fix
join-schema(xx, col1, col1, col2)
+--child1
+--child2 (col1,col1,col2)

1: If join's col1 and col1 leverage normal resolve index to do the column index location, it will both redirect to child2's first col1.

2: And join's inline projection will reversely mark `lused` nad `rused` from join itself to its children, a kind of flag mark array, and fill these cols into join chunk result when onMatch or not.

3: So if both come on, the join output chunk result will miss one column size, since filling result could't see the second col2 from flag array `rused` here

4: that's why we redirect the second same col1 in the join schema to the second same col1 in the right child.

5: but we need to notice one case here, we assume that two column sets are all ordered. That means the join schema must occur ordered from the merged schema.
One bad case is about:
```
join schema 里面是 [xx, col2, col1] 下面 limit 承载了 schema [col2 ,col1], 孩子是 sort，其孩子是 proj[col2, col1], datasource是[col1, col2].   
1: proj 留下 replace map 后消除了，limit 承载了 join 的 schema 即使使用 replace 自己替换自己看起来暂时ok，

2: pushTopN down，将 limit 和 sort 整合成 topN，topN 没有 schema！也就是这里回丢失 limit 的 schema。然后 join 直接到孩子 datasource 的 schema[col1, col2].

3: 然后现在就是 join [xx, col2, col1]，右侧孩子 topN 无schema，datasouce[col1, col2]

总结：join schema 的出现，不再根据左右孩子出现的顺序具有一定的偏序关系。遍历的时候可能需要一定的向后寻找才能找到
```

conclusion: The appearance of join schema no longer has a certain partial order relationship based on the order in which the left and right children appear. When traversing, you may need to search backwards to find it.